### PR TITLE
chore: set up license-eye to check license headers

### PR DESCRIPTION
**Pull request by kezhenxu94**
## What is the purpose of the pull request

This PR sets up license-eye to check the license headers, and add review comments when there are files missing license header.

## Brief change log

`.github/workflows/*.yml`:
  - Set up license-eye to check license headers.
  - Use built-in feature to checkout submodules instead of manually checking out.

## Verify this pull request

- [x] This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Comment by kezhenxu94** [comment]
@dailidong @lgcareer can you please take a look?

**Comment by chengshiwen** [comment]
@kezhenxu94 Maybe you should pass the unit test at first.

**Comment by kezhenxu94** [comment]
> @kezhenxu94 Maybe you should pass the unit test at first.

Check the changed files, I didn't touch the codes, the test case itself is flaky. Rerunning may help but I don't have access right to rerun.

**Comment by dailidong** [comment]
> > @kezhenxu94 Maybe you should pass the unit test at first.
> 
> Check the changed files, I didn't touch the codes, the test case itself is flaky. Rerunning may help but I don't have access right to rerun.

good job, I re-run the UT test

**Comment by kezhenxu94** [comment]
Still failed.

**Comment by CalvinKirs** [comment]
> Still failed.

Can you try to update your code? I already fixed the UT error.

**Comment by kezhenxu94** [comment]
> > Still failed.
> 
> Can you try to update your code? I already fixed the UT error.

Updated

**Comment by kezhenxu94** [comment]
@dailidong @CalvinKirs tests passed

File: pom.xml [hunk 1]
```diff
@@ -924,49 +923,6 @@
...
-                        <exclude>**/*.txt</exclude>
-                        <exclude>**/docs/**</exclude>
-                        <exclude>**/*.babelrc</exclude>
-                        <exclude>**/*.eslint*</exclude>
-                        <exclude>**/.mvn/jvm.config</exclude>
-                        <exclude>**/.mvn/wrapper/**</exclude>
-                    </excludes>
-                    <consoleOutput>true</consoleOutput>
-                </configuration>
-            </plugin>
```
Comment (CalvinKirs) [inline pom.xml:969]: Would it be better to keep the rat plugin? It can help us to conduct self-checks more conveniently in the development environment

Comment (kezhenxu94) [reply pom.xml:969]: Self checking is convenient with license-eye via

```shell
docker run -it --rm -v $(pwd):/github/workspace apache/skywalking-eyes header check
```

and if you have files missing license headers, you can use

```
docker run -it --rm -v $(pwd):/github/workspace apache/skywalking-eyes header fix
```

to add the headers automatically for you, which RAT plugin can't.

If you don't like using docker, you can still [download the binary](https://dist.apache.org/repos/dist/release/skywalking/eyes/0.1.0/skywalking-license-eye-0.1.0-bin.tgz) and run `license-eye header check/fix` to perform the checks/fixes.

However, if you want me to keep RAT plugin, I'm OK to do so, though I don't think it's necessary to keep it as license-eye has covered what RAT has

Comment (CalvinKirs) [reply pom.xml:969]: Deeply thanks for your answers. You are right

**Review by dailidong — APPROVED** [review]
+1
good job

**Pull request merged**